fed/mosaic-sumo/src/main/java/org/eclipse/mosaic/fed/sumo/ambassador/SumoAmbassador.java

```diff
@@ -420,36 +439,44 @@ public void finishSimulation() throws InternalFederateException {
     }
 
     /**
-     * Writes a new SUMO route file based on the registered vehicle types and routes.
+     * Writes a new SUMO route file based on the registered vehicle types.
      *
-     * @param typesInit  Interaction contains predefined vehicle types.
-     * @param routesInit Interaction contains paths and their IDs.
+     * @param typesInit Interaction contains predefined vehicle types.
      */
-    private void writeRouteFile(VehicleTypesInitialization typesInit, VehicleRoutesInitialization routesInit) {
-        File outputFile = initRouteFileCreator();
-
-        this.initialTypes.putAll(typesInit.getTypes());
+    private void writeTypesFromMapping(VehicleTypesInitialization typesInit) {
+        File outputFile = initVehicleTypeRouteFileCreator();
 
         // stores the rou.xml file to the working directory. this file is required for SUMO to run
         sumoRouteFileCreator
                 .addVehicleTypes(typesInit.getTypes())
-                .addRoutes(routesInit.getRoutes())
                 .store(outputFile);
     }
 
-    private File initRouteFileCreator() {
+    private File initVehicleTypeRouteFileCreator() {
         File dir = new File(descriptor.getHost().workingDirectory, descriptor.getId());
         String subDir = new File(sumoConfig.sumoConfigurationFile).getParent();
         if (StringUtils.isNotBlank(subDir)) {
             dir = new File(dir, subDir);
         }
-        File tmpRouteFile = new File(dir, routeFilename);
-
+        File tmpRouteFile = new File(dir, routeFileName);
+        File tmpVehicleTypeRouteFile = new File(dir, vehicleTypeRouteFileName);
+        File tmpSumoFile = new File(dir, sumoConfigurationFile.getName());
 
         // keep single instance
         if (sumoRouteFileCreator == null) {
-            this.sumoRouteFileCreator = new SumoRouteFileCreator(tmpRouteFile, sumoConfig.timeGapOffset);
+            sumoRouteFileCreator = new SumoRouteFileCreator(
+                    tmpSumoFile, tmpVehicleTypeRouteFile, sumoConfig.additionalVTypeParameters, sumoConfig.timeGapOffset
+            );
+            if (sumoConfig.writeVehicleDepartures) {
+                sumoRouteFileCreator.initializeDepartureDocument();
+            }
+        }
+        // FIXME: This seems a little bit dirty, but we configure our sumocfg's with an route-file even if it doesn't exist
```

Comment:
This could be fixed if we remove the route file definitions in the `.sumocfg`s where there doesn't already exist one (e.g. `HelloWorld` scenario)

Reply:
I do not get the proposed solution, can you give more details?

Reply:
Sumo needs a route file to start (I think).  
Right now all of our `.sumocfg`s link to a route file, even though this file doesn't have to exist. Because with the new `SumoRouteFileCreator` we edit the `.sumocfg` automatically (adding `..._vTypes.rou.xml`) we could remove this non-existent route file.
See below for example:
```xml
...
<input>
    <net-file value="highway.net.xml" />
    <route-files value="highway.rou.xml" />
</input>
...
```
||
V
```xml
...
<input>
    <net-file value="highway.net.xml" />
    <route-files/>
</input>
...
```
OR
```xml
...
<input>
    <net-file value="highway.net.xml" />
    <route-files value=""/>
</input>
...
```

Reply:
If we adapt scenarios, which don't have a route file, we could omit adding an empty route file

Reply:
can we remove the route-files tag completely? :)

Reply:
Yeah we could also do that. That requires a little bit more XML-handling, which shouldn't make the code much more ugly.

So I have the "GO" to adapt all scenarios according to the proposed change?

Reply:
Go for it